app/src/main/java/com/dongyang/android/youdongknowme/ui/view/search/SearchViewModel.kt

```diff
@@ -1,6 +1,17 @@
 package com.dongyang.android.youdongknowme.ui.view.search
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import com.dongyang.android.youdongknowme.standard.base.BaseViewModel
 
 class SearchViewModel : BaseViewModel() {
+    private val _searchContent: MutableLiveData<String> = MutableLiveData()
+    val searchContent: MutableLiveData<String> = _searchContent
+
+    private val _searchClearVisibility: MutableLiveData<Boolean> = MutableLiveData()
+    val searchClearVisibility: LiveData<Boolean> = _searchClearVisibility
+
+    fun validateSearchClearButtonVisibility() {
+        _searchClearVisibility.value = !_searchContent.value.isNullOrEmpty()
+    }
```

Comment:
```suggestion
    fun validateSearchClearButtonVisibility() {
        _searchClearVisibility.value = _searchContent.value.isNullOrEmpty().not()
    }
```

`not()` 을 사용하면 가독성이 좀 더 좋아져요!